src/pg/pg-store.ts

```diff
@@ -212,6 +212,36 @@ export class PgStore extends BasePgStore {
     return parseInt(result[0].block_height);
   }
 
+  /**
+   * Returns the block hash of the latest block, or the block hash of the block
+   * at the given height.
+   * @param blockHeight - optional block height (defaults to latest block)
+   */
+  async getBlockHash(blockHeight?: string): Promise<string> {
+    const clause = blockHeight
+      ? this.sql`WHERE block_height = ${blockHeight}`
+      : this.sql`
+        ORDER BY block_height DESC
+        LIMIT 1
+      `;
+
+    const result = await this.sql<{ block_hash: string }[]>`
+      SELECT block_hash FROM inscriptions_per_block
```

Comment:
I think we could make these queries more efficient if we modified the `chain_tip` materialized view to keep also the block hash and timestamp. That way it would be a O(1) retrieval from the DB always.